libs/portal-integration-angular/src/lib/core/components/button-dialog/button-dialog.component.ts

```diff
@@ -0,0 +1,210 @@
+import {
+  Component,
+  ComponentRef,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core'
+import { Observable, from, isObservable, map, of, startWith } from 'rxjs'
+import { DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog'
+
+import { ButtonDialogButtonDetails, ButtonDialogConfig, ButtonDialogData } from '../../../model/button-dialog'
+import { DialogMessageContentComponent } from './dialog-message-content/dialog-message-content.component'
+import {
+  DialogButtonClicked,
+  DialogPrimaryButtonDisabled,
+  DialogResult,
+  DialogSecondaryButtonDisabled,
+  DialogState,
+} from '../../../services/portal-dialog.service'
+
+@Component({
+  selector: 'ocx-button-dialog',
+  templateUrl: './button-dialog.component.html',
+})
+export class ButtonDialogComponent implements OnInit {
+  defaultPrimaryButtonDetails: ButtonDialogButtonDetails = {
+    key: 'Confirm',
```

Comment:
The default keys should be prefixed with the name of the component (see translation files). Please also add transitions to the translation files.